docs/architecture/adr-003-ics27-acknowledgement.md

```diff
@@ -0,0 +1,118 @@
+
+## Status
+
+Accepted
+
+## Context
+
+Upon receiving an IBC packet, an IBC application can optionally return an acknowledgement. 
+This acknowledgement will be hashed and written into state. Thus any changes to the information included in an acknowledgement are state machine breaking. 
+
+ICS27 executes transactions on behalf of a controller chain. Information such as the message result or message error may be returned from other SDK modules outside the control of the ICS27 module. 
+It might be very valuable to return message execution information inside the ICS27 acknowledgement so that controller chain interchain account auth modules can act upon this information. 
+Only determinstic information returned from the message execution is allowed to be returned in the packet acknowledgement otherwise the network will halt due to a fork in the expected app hash. 
+
+## Decision
+
+At the time of this writing, Tendermint includes the following information in the [ABCI.ResponseDeliverTx](https://github.com/tendermint/tendermint/blob/release/v0.34.13/types/results.go#L47-#L53):
+```go
+// deterministicResponseDeliverTx strips non-deterministic fields from
+// ResponseDeliverTx and returns another ResponseDeliverTx.
+func deterministicResponseDeliverTx(response *abci.ResponseDeliverTx) *abci.ResponseDeliverTx {
+	return &abci.ResponseDeliverTx{
+		Code:      response.Code,
+		Data:      response.Data,
+		GasWanted: response.GasWanted,
+		GasUsed:   response.GasUsed,
+	}
+}
+```
+
+### Successful acknowledgements
+
+Successful acknowledgements should return information about the transaction execution. 
+Given the determinstic fields in the `abci.ResponseDeliverTx`, the transaction `Data` can be used to indicate information about the transaction execution. 
+The `abci.ResponseDeliverTx.Data` will be set in the ICS27 packet acknowledgement upon successful transaction execution.
+
+The format for the `abci.ResponseDeliverTx.Data` is constructed by the SDK. 
+
+At the time of this writing, the next major release of the SDK will change the format for constructing the transaction response data. 
+
+#### v0.45 format
+
+The current version, v0.45 constructs the transaction response as follows:
+```go
+    proto.Marshal(&sdk.TxMsgData{
+        Data: []*sdk.MsgData{msgResponses...}, 
+    }
+```
+
+Where `msgResponses` is a slice of `*sdk.MsgData`. 
+The `MsgData.MsgType` contains the `sdk.MsgTypeURL` of the `sdk.Msg` being executed.
+The `MsgData.Data` contains the proto marshaled `MsgResponse` for the associated message executed. 
+
+#### Next major version format
+
+The next major version will construct the transaction response as follows:
+```go 
+    proto.Marshal(&sdk.TxMsgData{
+        MsgResponses: []*codectypes.Any{msgResponses...}, 
+    }
+```
+
+Where `msgResponses` is a slice of the `MsgResponse`s packed into `Any`s.
+
+#### Forwards compatible approach
+
+A forwards compatible approach was deemed infeasible. 
```

Comment:
I think the current suggestion works for the short term. Great ideas.  

Should we contact the SDK team to perhaps come up with a future-proof plan for this functionality? Maybe there is some format we can come up with for IBC application developers that makes sense. Definitely doesn't need to be done right away.

cc: @crodriguezvega

Reply:
I don't think the SDK will change the format for a long time after this next change. I also think returning a slice of `Any`s is a great design choice. I don't see any limitations in the decision

Reply:
Cool, makes sense, I guess I more so just meant that the team is aware any changes would affect us as consumers.